web_widget_text_count/static/src/js/fields_text_count.js

```diff
@@ -0,0 +1,50 @@
+odoo.define('text_count', function(require) {
+    "use strict";
+
+    var registry = require('web.field_registry');
+    var basicFields = require('web.basic_fields');
+    var FieldText = basicFields.FieldText;
+
+    var FieldTextCount = FieldText.extend({
+        events: _.extend({}, FieldText.prototype.events, {
+            'input': 'count_char',
+        }),
+        supportedFieldTypes: ['text'],
+
+        start: function () {
+            var self = this;
+            return this._super().then(function(){
+                var size = 0
+                if (self.mode === 'edit') {
+                    if (self.attrs.size) {
+                        self.$el.attr('maxlength', parseInt(self.attrs.size));
+                        size = self.attrs.size - self.$el.val().length;
+                    }
+                    else{
+                        size = self.$el.val().length;
+                    }
+                    self.$el = self.$el.add($('<input class="text-counter-widget" readonly="readonly" value="'+size+'"/>'));
```

Comment:
Why an `input` and not a `span`  with `o_readonly_modifier` class? The counter looks like it is accepting input when in fact it does not.